test/box-luatest/gh_6011_follow_env_vars_on_log_cfg_test.lua

```diff
@@ -0,0 +1,49 @@
+local t = require('luatest')
+local g = t.group()
+
+local TARANTOOL_PATH = arg[-1]
+
+-- Test 'level' default value so that `test_env_set` success is not because
+-- of 'warn' default.
+local test_default = [[
+    log = require('log')
+    log.cfg{}
+    os.exit(log.cfg.level == 5 and 0 or 1)
+]]
+
+-- Main test. Test env vars are taken into account in log.cfg().
+local test_env_set = [[
+    log = require('log')
+    log.cfg{}
+    os.exit(log.cfg.level == 'warn' and 0 or 1)
+]]
+
+-- Test we don't apply env vars again in box.cfg() after log.cfg for
+-- log options.
+local test_box_env_dont_overwrite = [[
+    log = require('log')
+    log.cfg{level='error'}
+    box.cfg{}
+    os.exit(log.cfg.level == 'error' and 0 or 1)
+]]
+
+-- Test we don't fail in log.cfg() if non log options are specified
+-- incorrectly in env vars.
+local test_other_env_option_error = [[
+    log = require('log')
+    log.cfg{level='error'}
+    os.exit(log.cfg.level == 'error' and 0 or 1)
+]]
+
+local function run_test(test, env)
+    env = env or ""
+    local cmd = string.format('%s %s -e "%s"', env, TARANTOOL_PATH, test)
+    t.assert_equals(os.execute(cmd), 0)
```

Comment:
Added `fio.chdir(server.vardir)` at the beginnig of the test to keep tarantool products there.

Reply:
You shouldn't pollute the var directory, because other tests may run before/after this one. Better create a temporary directory named after the test (e.g. `gh-6011-data`) in the var directory and delete it when the test is complete. Also, using `fio.chdir` in luatest is bad, because luatest may run other tests. Better do it in the started script.

Reply:
Done.